libs/utils/wa_results_collector.py

```diff
@@ -484,9 +484,11 @@ def _get_extra_job_metrics(self, job_dir, workload):
                     df.loc[:, 'metric'] = 'device_power_sample'
                     df.loc[:, 'units'] = 'watts'
 
-                    metrics_df = metrics_df.append(df)
-
-        return metrics_df
+                    extra_metric_list.append(df)
+        if len(extra_metric_list) > 0:
```

Comment:
What about keeping the empty DF initialization at the beginning and replace these two lines with just:
```python
if len(extra_metric_list):
    metrics_df = metrics_df.append(extra_metric_list)
return metrics_df
```
nit-pick: I would also add an empty line before the if check.